TravelAgency-spring-mvc/src/main/webapp/WEB-INF/jsp/customer/edit.jsp

```diff
@@ -40,17 +41,21 @@
                 </tr>
                 <tr>
                     <td><form:label path="password"><f:message key="customer.password"/>: </form:label></td>
-                    <td><form:input path="password"/></td>
+                    <td><form:input type="password" path="password"/></td>
                 </tr>
-                <br/> 
+                <br/>
+                <br/>
                 <br/>
                 <tr>
                     <td><p class="text-center">
-                            <button class="btn btn-primary" type="submit">Update</button>
-                        </p></td>
+                        <button class="btn btn-primary" type="submit">Update</button>
+                    </p></td>
                     <td><p class="text-center">
-                            <button class="btn btn-danger" type="submit">Delete</button>
-                        </p></td>
+                           <my:a href="/customer/delete/${customer.id}" class="btn btn-danger">
```

Comment:
This is not really a safe approach. Now anyone could delete any customer by putting id to the address. I suggest obtaining the ID from the authenticated user by the controller.